File: 1127-last-stone-weight/last-stone-weight.cs

public class MaxHeapComparer: IComparer<int>
{
    public int Compare(int x, int y){
        return y.CompareTo(x);
    }
}


public class Solution {
    public int LastStoneWeight(int[] stones) {
        var pq = new PriorityQueue<int,int>(Comparer<int>.Create((x,y)=>y.CompareTo(x)));
        for(int i = 0 ; i < stones.Length ; i++){
            pq.Enqueue(stones[i], stones[i]);
        }
        while(pq.Count > 1){
            int y = pq.Dequeue();
            int x = pq.Dequeue();
            if(y > x)
                pq.Enqueue(y-x, y-x);
        }
        return pq.Count != 0 ? pq.Dequeue() : 0;
    }
}

File: 1127-last-stone-weight/last-stone-weight.java
import java.util.PriorityQueue;
import java.util.Arrays;
import java.util.List;
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>(Collections.reverseOrder());
        for(int i = 0 ; i < stones.length ; i++){
            maxHeap.add(stones[i]);
        }
        while(maxHeap.size() > 1){
            int y = maxHeap.remove();
            int x = maxHeap.remove();
            if(y > x){
                maxHeap.add(y-x);
            } 
        }
        return maxHeap.size() == 1 ? maxHeap.remove() : 0;
    }
}